prov/verbs/src/fi_verbs.h

```diff
@@ -162,6 +173,16 @@ struct fi_ibv_cq {
 	enum fi_cq_wait_cond	wait_cond;
 	struct ibv_wc		wc;
 	int			signal_fd[2];
+	fi_ibv_cq_read_entry	read_entry;
+	struct util_buf_pool	*wce_pool;
+	struct slist		wcq;
+	fastlock_t		lock;
+	struct util_buf_pool	*epe_pool;
+	struct slist		ep_list;
+	fastlock_t		ep_list_lock;
+	uint64_t		ep_cnt;
+	uint64_t		send_signal_wr_id;
+	uint64_t		wr_id_mask;
```

Comment:
I guess moving the pools to the fabric would be a good idea. Initially, I started with two locks to serialize access between two different lists (ep_list and wcq). Later, I expanded the wcq lock coverage to include polling too (to keep things simple). The only place where ep_list_lock is used and cq->lock is not is when we add an EP to ep_list. I can replace the two with one lock if it's not gonna affect performance much.